lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -892,8 +893,25 @@ bool DeclVisitor::handleStructType(const clang::RecordDecl *RD) {
     CurrentOffset += *Size;
   }
 
-  // TODO: Can this be calculated/fetched automatically?
-  Struct->Size() = CurrentOffset;
+  std::optional ExplicitSize = parseIntegerAnnotation(*Definition,
+                                                      SizeAnnotation);
+  if (not ExplicitSize.has_value()) {
+    revng_log(Log, "Explicit struct size tag is missing.");
+    // TODO: do we actually want to allow this through???
```

Comment:
While editing, the user should still be able to drop `_SIZE`, `_START_AT` and use padding fields.
Are we supporting both modes of operation?

Reply:
This comment is specifically my doubt about whether we want size-less structs.

As for padding, yes, both modes can be used interchangeably. The only forbidden thing is `_START_AT` on an explicit padding field.

Reply:
It's not clear to me if we allow both modes at the same time on a single struct, or it must be either one or the other.

Reply:
To go back to your question @Cvelth: do we want size-less structs?

I think we do. The scenario I have in mind is: a user knows that some offsets in memory there's some data with a given type. They want to wrap it into a struct, and they don't want to be forced to manually compute it's size only to use it in the annotation. They want to just be able to tell `revng`: here's this struct, with these fields inside, and you figure out the minimum size needed to fit.

Reply:
Yeah, I think we are on the same page as far as that is concerned. The check and TODO above are dropped already.